Utilities/XrdAdaptor/src/QualityMetric.h

```diff
@@ -66,7 +66,7 @@ friend class Source;
 
     static QualityMetricFactory *m_instance;
 
-    typedef std::unordered_map<std::string, QualityMetricUniqueSource*> MetricMap;
+    typedef tbb::concurrent_hash_map<std::string, QualityMetricUniqueSource*> MetricMap;
```

Comment:
I think it would be better to use ```tbb::concurrent_unordered_map<std::string, std::shared_ptr<QualityMetricUniqueSource>>``` as the type. That type is guaranteed to take no visible locks. ```std::concurrent_hash_map``` is preferred if one need concurrent erasure, which is not the case here.

https://www.threadingbuildingblocks.org/docs/help/reference/containers_overview/concurrent_unordered_map_cls.htm